pilot/pkg/model/telemetry.go

```diff
@@ -1052,3 +1056,56 @@ func generateStatsConfig(class networking.ListenerClass, metricsCfg telemetryFil
 func disableHostHeaderFallback(class networking.ListenerClass) bool {
 	return class == networking.ListenerClassSidecarInbound || class == networking.ListenerClassGateway
 }
+
+// resourceAttributes return semantic attributes follow OpenTelemetry specification
+// see more details here: https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/
+func resourceAttributes(proxy *Proxy) *otlpcommon.KeyValueList {
+	podName, _ := extractProxyID(proxy)
+	canonicalName := proxy.Labels[IstioCanonicalServiceLabelName]
+	canonicalRevision := proxy.Labels[IstioCanonicalServiceRevisionLabelName]
+	return &otlpcommon.KeyValueList{
+		Values: []*otlpcommon.KeyValue{
+			otelKeyValue(otelsemconv.K8SClusterNameKey, proxy.Metadata.ClusterID.String()),
+			otelKeyValue(otelsemconv.K8SNamespaceNameKey, proxy.ConfigNamespace),
+			otelKeyValue(otelsemconv.K8SDeploymentNameKey, proxy.Metadata.WorkloadName),
+			otelKeyValue(otelsemconv.K8SPodNameKey, podName),
+			otelKeyValue(otelsemconv.ServiceNameKey, canonicalName),
+			otelKeyValue(otelsemconv.ServiceVersionKey, canonicalRevision),
+		},
+	}
+}
+
+func otelKeyValue(semantic attribute.Key, val string) *otlpcommon.KeyValue {
+	return &otlpcommon.KeyValue{
+		Key:   string(semantic),
+		Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: val}},
+	}
+}
+
+func extractProxyID(proxy *Proxy) (string, string) {
+	// ID is the unique platform-specific sidecar proxy ID. For k8s it is the pod ID and
+	// namespace <podName.namespace>.
+	ids := strings.Split(proxy.ID, ".")
+	if len(ids) < 2 {
+		// should not happen, just incase
+		return proxy.ID, ""
+	}
+	return ids[0], ids[1]
+}
+
+func Baggage(proxy *Proxy) string {
+	attrs := resourceAttributes(proxy)
+
+	members := make([]otelbaggage.Member, 0, len(attrs.Values)-1)
+	for _, attr := range attrs.Values {
+		if attr.Key == string(otelsemconv.K8SPodNameKey) {
+			// k8s.pod.name is unneeded in baggage
```

Comment:
Why are we inconsistent in baggage and attributes?

Reply:
just keep the baggage output same as before, I think it's reasonable to pass `pod.name` in opentelemetry log, but I'm not sure of `baggage` header, it's meaningless for now.